jobs/mongodb_migration/src/mongodb_migration/migrations/_20240112164500_cache_add_partial_field_in_split_descriptive_statistics.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 The HuggingFace Authors.
+
+import logging
+
+from libcommon.constants import CACHE_COLLECTION_RESPONSES, CACHE_MONGOENGINE_ALIAS
+from libcommon.parquet_utils import parquet_export_is_partial
+from libcommon.simple_cache import CachedResponseDocument
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddPartialToSplitDescriptiveStatisticsCacheResponse(Migration):
+    def up(self) -> None:
+        # See https://docs.mongoengine.org/guide/migration.html#example-1-addition-of-a-field
+        logging.info(
+            "If missing, add the 'partial' field with the default value None"
+            " to the cached results of split-descriptive-statistics job runner"
+        )
+        db = get_db(CACHE_MONGOENGINE_ALIAS)
+        partial_configs_entries = db[CACHE_COLLECTION_RESPONSES].find(
+            {
+                "kind": "config-parquet",
+                "content.partial": True,
```

Comment:
Maybe should filter `"http_status": 200` so that it can help speed up the query.